fuzzy.go

```diff
@@ -55,35 +54,38 @@ characters upto the first match.
 */
 func Find(pattern string, data []string) []Match {
 	if len(pattern) == 0 {
-		return []Match{}
+		return nil
```

Comment:
I guess it would be easier for client code if this just returned an empty array like before.

Reply:
In what case is an empty slice more usable than a nil slice? Note, none of the common things you do with a slice panics if the slice is nil ([see playground](https://play.golang.org/p/lgusiNOG3LJ)).

Reply:
I have no strong preferences here. Go authors [prefer](https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices) nil slices so let's go with returning a nil slice.